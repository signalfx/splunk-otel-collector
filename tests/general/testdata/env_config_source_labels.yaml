config_sources:
  env:

receivers:
  hostmetrics:
    collection_interval: 2s
    scrapers:
      memory:
processors:
  filter:
    metrics:
      include:
        match_type: strict
        metric_names:
          - system.memory.usage
  metricstransform:
    transforms:
      - action: update
        include: .*
        match_type: regexp
        operations:
          - action: add_label
            new_label: single-dollar
            new_value: ${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: single-dollar-no-curly-braces
            new_value: prefix-$env:AN_ENVVAR
          - action: add_label
            new_label: double-dollar
            new_value: $${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: double-dollar-no-curly-braces
            new_value: prefix-$$env:AN_ENVVAR
          - action: add_label
            new_label: triple-dollar
            new_value: $$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: triple-dollar-no-curly-braces
            new_value: prefix-$$$env:AN_ENVVAR
          - action: add_label
            new_label: quadruple-dollar
            new_value: $$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: quadruple-dollar-no-curly-braces
            new_value: prefix-$$$$env:AN_ENVVAR
          - action: add_label
            new_label: quintuple-dollar
            new_value: $$$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: quintuple-dollar-no-curly-braces
            new_value: prefix-$$$$$env:AN_ENVVAR
          - action: add_label
            new_label: sextuple-dollar
            new_value: $$$$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: sextuple-dollar-no-curly-braces
            new_value: prefix-$$$$$$env:AN_ENVVAR
          - action: add_label
            new_label: septuple-dollar
            new_value: $$$$$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: septuple-dollar-no-curly-braces
            new_value: prefix-$$$$$$$env:AN_ENVVAR
          - action: add_label
            new_label: octuple-dollar
            new_value: $$$$$$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: octuple-dollar-no-curly-braces
            new_value: prefix-$$$$$$$$env:AN_ENVVAR
          - action: add_label
            new_label: nonuple-dollar
            new_value: $$$$$$$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: nonuple-dollar-no-curly-braces
            new_value: prefix-$$$$$$$$$env:AN_ENVVAR
          - action: add_label
            new_label: decuple-dollar
            new_value: $$$$$$$$$${env:AN_ENVVAR}-suffix
          - action: add_label
            new_label: decuple-dollar-no-curly-braces
            new_value: prefix-$$$$$$$$$$env:AN_ENVVAR

exporters:
  otlp:
    endpoint: "${OTLP_ENDPOINT}"
    tls:
      insecure: true
service:
  pipelines:
    metrics:
      receivers:
        - hostmetrics
      processors:
        - filter
        - metricstransform
      exporters:
        - otlp
