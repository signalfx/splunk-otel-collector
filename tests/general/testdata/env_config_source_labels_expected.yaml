resourceMetrics:
  - resource: {}
    schemaUrl: https://opentelemetry.io/schemas/1.9.0
    scopeMetrics:
      - metrics:
          - description: Bytes of memory in use.
            name: system.memory.usage
            sum:
              aggregationTemporality: 2
              dataPoints:
                - asInt: "224563200"
                  attributes:
                    - key: decuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: decuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$$env:AN_ENVVAR
                    - key: double-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: double-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$env:AN_ENVVAR
                    - key: nonuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: nonuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$an-envvar-value
                    - key: octuple-dollar
                      value:
                        stringValue: $$$${env:AN_ENVVAR}-suffix
                    - key: octuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$env:AN_ENVVAR
                    - key: quadruple-dollar
                      value:
                        stringValue: $${env:AN_ENVVAR}-suffix
                    - key: quadruple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$env:AN_ENVVAR
                    - key: quintuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: quintuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$an-envvar-value
                    - key: septuple-dollar
                      value:
                        stringValue: $$$an-envvar-value-suffix
                    - key: septuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$an-envvar-value
                    - key: sextuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: sextuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$env:AN_ENVVAR
                    - key: single-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: single-dollar-no-curly-braces
                      value:
                        stringValue: prefix-an-envvar-value
                    - key: state
                      value:
                        stringValue: buffered
                    - key: triple-dollar
                      value:
                        stringValue: $an-envvar-value-suffix
                    - key: triple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$an-envvar-value
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "4355174400"
                  attributes:
                    - key: decuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: decuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$$env:AN_ENVVAR
                    - key: double-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: double-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$env:AN_ENVVAR
                    - key: nonuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: nonuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$an-envvar-value
                    - key: octuple-dollar
                      value:
                        stringValue: $$$${env:AN_ENVVAR}-suffix
                    - key: octuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$env:AN_ENVVAR
                    - key: quadruple-dollar
                      value:
                        stringValue: $${env:AN_ENVVAR}-suffix
                    - key: quadruple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$env:AN_ENVVAR
                    - key: quintuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: quintuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$an-envvar-value
                    - key: septuple-dollar
                      value:
                        stringValue: $$$an-envvar-value-suffix
                    - key: septuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$an-envvar-value
                    - key: sextuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: sextuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$env:AN_ENVVAR
                    - key: single-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: single-dollar-no-curly-braces
                      value:
                        stringValue: prefix-an-envvar-value
                    - key: state
                      value:
                        stringValue: cached
                    - key: triple-dollar
                      value:
                        stringValue: $an-envvar-value-suffix
                    - key: triple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$an-envvar-value
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "3182825472"
                  attributes:
                    - key: decuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: decuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$$env:AN_ENVVAR
                    - key: double-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: double-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$env:AN_ENVVAR
                    - key: nonuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: nonuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$an-envvar-value
                    - key: octuple-dollar
                      value:
                        stringValue: $$$${env:AN_ENVVAR}-suffix
                    - key: octuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$env:AN_ENVVAR
                    - key: quadruple-dollar
                      value:
                        stringValue: $${env:AN_ENVVAR}-suffix
                    - key: quadruple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$env:AN_ENVVAR
                    - key: quintuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: quintuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$an-envvar-value
                    - key: septuple-dollar
                      value:
                        stringValue: $$$an-envvar-value-suffix
                    - key: septuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$an-envvar-value
                    - key: sextuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: sextuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$env:AN_ENVVAR
                    - key: single-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: single-dollar-no-curly-braces
                      value:
                        stringValue: prefix-an-envvar-value
                    - key: state
                      value:
                        stringValue: free
                    - key: triple-dollar
                      value:
                        stringValue: $an-envvar-value-suffix
                    - key: triple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$an-envvar-value
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "250474496"
                  attributes:
                    - key: decuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: decuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$$env:AN_ENVVAR
                    - key: double-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: double-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$env:AN_ENVVAR
                    - key: nonuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: nonuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$an-envvar-value
                    - key: octuple-dollar
                      value:
                        stringValue: $$$${env:AN_ENVVAR}-suffix
                    - key: octuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$env:AN_ENVVAR
                    - key: quadruple-dollar
                      value:
                        stringValue: $${env:AN_ENVVAR}-suffix
                    - key: quadruple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$env:AN_ENVVAR
                    - key: quintuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: quintuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$an-envvar-value
                    - key: septuple-dollar
                      value:
                        stringValue: $$$an-envvar-value-suffix
                    - key: septuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$an-envvar-value
                    - key: sextuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: sextuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$env:AN_ENVVAR
                    - key: single-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: single-dollar-no-curly-braces
                      value:
                        stringValue: prefix-an-envvar-value
                    - key: state
                      value:
                        stringValue: slab_reclaimable
                    - key: triple-dollar
                      value:
                        stringValue: $an-envvar-value-suffix
                    - key: triple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$an-envvar-value
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "73023488"
                  attributes:
                    - key: decuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: decuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$$env:AN_ENVVAR
                    - key: double-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: double-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$env:AN_ENVVAR
                    - key: nonuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: nonuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$an-envvar-value
                    - key: octuple-dollar
                      value:
                        stringValue: $$$${env:AN_ENVVAR}-suffix
                    - key: octuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$env:AN_ENVVAR
                    - key: quadruple-dollar
                      value:
                        stringValue: $${env:AN_ENVVAR}-suffix
                    - key: quadruple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$env:AN_ENVVAR
                    - key: quintuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: quintuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$an-envvar-value
                    - key: septuple-dollar
                      value:
                        stringValue: $$$an-envvar-value-suffix
                    - key: septuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$an-envvar-value
                    - key: sextuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: sextuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$env:AN_ENVVAR
                    - key: single-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: single-dollar-no-curly-braces
                      value:
                        stringValue: prefix-an-envvar-value
                    - key: state
                      value:
                        stringValue: slab_unreclaimable
                    - key: triple-dollar
                      value:
                        stringValue: $an-envvar-value-suffix
                    - key: triple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$an-envvar-value
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "4770893824"
                  attributes:
                    - key: decuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: decuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$$env:AN_ENVVAR
                    - key: double-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: double-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$env:AN_ENVVAR
                    - key: nonuple-dollar
                      value:
                        stringValue: $$$$an-envvar-value-suffix
                    - key: nonuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$an-envvar-value
                    - key: octuple-dollar
                      value:
                        stringValue: $$$${env:AN_ENVVAR}-suffix
                    - key: octuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$$env:AN_ENVVAR
                    - key: quadruple-dollar
                      value:
                        stringValue: $${env:AN_ENVVAR}-suffix
                    - key: quadruple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$env:AN_ENVVAR
                    - key: quintuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: quintuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$an-envvar-value
                    - key: septuple-dollar
                      value:
                        stringValue: $$$an-envvar-value-suffix
                    - key: septuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$an-envvar-value
                    - key: sextuple-dollar
                      value:
                        stringValue: $$an-envvar-value-suffix
                    - key: sextuple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$$$env:AN_ENVVAR
                    - key: single-dollar
                      value:
                        stringValue: an-envvar-value-suffix
                    - key: single-dollar-no-curly-braces
                      value:
                        stringValue: prefix-an-envvar-value
                    - key: state
                      value:
                        stringValue: used
                    - key: triple-dollar
                      value:
                        stringValue: $an-envvar-value-suffix
                    - key: triple-dollar-no-curly-braces
                      value:
                        stringValue: prefix-$an-envvar-value
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
            unit: By
        scope:
          name: otelcol/hostmetricsreceiver/memory
          version: v0.97.0
